File: utilities/java/parse.java
import java.util.*;
import java.util.regex.Pattern;
import java.util.stream.Collectors;

/**
 * Parsing utilities for Java solutions.
 */
public class Parse {
    /**
     * Parse integers from a line of text.
     * @param line Line of text containing integers
     * @return List of integers found in the line
     */
    public static List<Integer> parseInts(String line) {
        Pattern pattern = Pattern.compile("-?\\d+");
        return pattern.matcher(line)
            .results()
            .map(m -> Integer.parseInt(m.group()))
            .collect(Collectors.toList());
    }
}

File: utilities/kotlin/parse.kt
import java.util.regex.Pattern
import java.util.regex.Matcher

/**
 * Parsing utilities for Kotlin solutions.
 */

/**
 * Parse integers from a line of text.
 */
fun parseInts(line: String): List<Int> {
    val pattern = Pattern.compile("-?\\d+")
    val matcher = pattern.matcher(line)
    val result = mutableListOf<Int>()
    while (matcher.find()) {
        result.add(matcher.group().toInt())
    }
    return result
}
